model/map-hot-rod/src/main/java/org/keycloak/models/map/storage/hotRod/IckleQueryMapModelCriteriaBuilder.java

```diff
@@ -172,7 +180,7 @@ private StringBuilder getWhereClauseBuilder() {
      * @return Ickle query that represents this QueryBuilder
      */
     public String getIckleQuery() {
-        return "FROM org.keycloak.models.map.storage.hotrod.HotRodClientEntity " + C + ((whereClauseBuilder.length() != 0) ? " WHERE " + whereClauseBuilder : "");
+        return "FROM " + HOT_ROD_ENTITY_PACKAGE + "." + hotRodEntityClass.getSimpleName() + " " + C + ((whereClauseBuilder.length() != 0) ? " WHERE " + whereClauseBuilder : "");
```

Comment:
Can the following be used?
```suggestion
        return "FROM " + hotRodEntityClass.getCanonicalName() + " " + C + ((whereClauseBuilder.length() != 0) ? " WHERE " + whereClauseBuilder : "");
```

Reply:
I believe this is not possible. As far as I know, Inifinispan proto definition generator doesn't consider [real class package](https://github.com/keycloak/keycloak/blob/2b0167db8f661c915125133926cfc3b0e768114e/model/map-hot-rod/src/main/java/org/keycloak/models/map/storage/hotRod/group/HotRodGroupEntity.java#L18) when creating definitions. We have just one proto definition [(example)](https://gist.github.com/mhajas/5da6570ece22e324165f41adf6f06d1e) generated with one package that is [specified here](https://github.com/keycloak/keycloak/blob/2b0167db8f661c915125133926cfc3b0e768114e/model/map-hot-rod/src/main/java/org/keycloak/models/map/storage/hotRod/common/ProtoSchemaInitializer.java#L45).

So this would be possible only if we would have all HotRodEntities (HotRodClientEntity, HotRodUserEntity, HotRodPair, etc.) in the same java package or if we generate one proto definition per java package. @martin-kanis Any other idea?

Reply:
In tat case, could the HOT_ROD_ENTITY_PACKAGE be shorter than is, like `kc.`?

Reply:
I think so, I will give it a try.